File: src/main/java/SymmetricTree/Zhanwentao/SymmetricTree.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao
from builtins import len, list, range, object


class TreeNode(object):
    def __init__(self, x,left,right):
        self.val = x
        self.left = left
        self.right = right

def solution(root) :
    if root is None or (root.left is None and root.right is None) :
        return True
    nodes = [root.left,root.right]
    while len(nodes) != 0:
      for i in range(len(nodes)/2):
          left = None if nodes[i] is None else nodes[i].val
          right = None if nodes[len(nodes) - i - 1] is None else nodes[len(nodes) - i - 1].val
          if left != right :
              return False

      tmp = list(nodes)
      nodes.clear()
      flag = True
      for node in tmp:
          if node != None:
              if node.left != None or node.right != None: flag = True
              nodes.append(node.left)
              nodes.append(node.right)
      if flag == False :
          break
    return True

File: src/main/java/SymmetricTree/jenson/SymmetricTree.java
package SymmetricTree.jenson;

/**
 * @author: zhousong
 * @data: 2019/6/17
 */
public class SymmetricTree {
    public boolean isSymmetric(TreeNode root) {
        if (root == null) {
            return true;
        }
        return isSymmetric(root.left, root.right);
    }

    private boolean isSymmetric(TreeNode left, TreeNode right) {
        if (left == null && right == null) {
            return true;
        } else if (left == null) {
            return false;
        } else if (right == null) {
            return false;
        }

        if (left.val != right.val) {
            return false;
        }

        return isSymmetric(left.left, right.right) && isSymmetric(left.right, right.left);
    }

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
}
